api/v1alpha1/gitops_types.go

```diff
@@ -0,0 +1,100 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// GitOpsSpec defines the desired state of GitOps configuration
+type GitOpsSpec struct {
+	// Overrides the default GitOps configuration with custom configuration for the specified app(s).
+	Configutaion []Configuration `json:"config"`
+}
+
+// CommitMode describes how Turbonomic events will be processed.
+// +kubebuilder:validation:Enum=direct;request
+type CommitMode string
```

Comment:
Please elaborate on the testing section.
Specifically, please list both positive and negative tests that you did for CRD and CR creation.

Reply:
Added additional information about some of the various test cases.